src/Internal/Plutip/Types.purs

```diff
@@ -95,21 +113,41 @@ data InitialUTxOsWithStakeKey =
 
 type InitialUTxODistribution = Array InitialUTxOs
 
-newtype ClusterStartupRequest = ClusterStartupRequest
-  { keysToGenerate :: InitialUTxODistribution
-  , epochSize :: UInt
-  , slotLength :: Seconds
-  }
+newtype ClusterStartupRequest =
+  ClusterStartupRequest
+    { keysToGenerate :: InitialUTxODistribution
+    , epochSize :: Maybe UInt
+    , slotLength :: Maybe Seconds
+    , maxTxSize :: Maybe UInt
+    , increasedExUnits :: Maybe UInt
+    , noCollateral :: Boolean
+    }
 
 instance EncodeAeson ClusterStartupRequest where
   encodeAeson
     ( ClusterStartupRequest
-        { keysToGenerate, epochSize, slotLength: Seconds slotLength }
-    ) = encodeAeson
-    { keysToGenerate
-    , epochSize
-    , slotLength: unsafePartial partialFiniteNumber slotLength
-    }
+        { keysToGenerate
+        , epochSize
+        , slotLength
+        , maxTxSize
+        , increasedExUnits
+        , noCollateral
+        }
+    ) =
+
+    let
+      sl = case slotLength of
+        Just (Seconds n) -> Just $ unsafePartial partialFiniteNumber n
+        Nothing -> Nothing
```

Comment:
we can simplify this:
```suggestion
      sl = slotLength >>= (finiteNumber <<< unwrap)
```

Reply:
[def59aa](https://github.com/Plutonomicon/cardano-transaction-lib/pull/1334/commits/def59aa1b3f6a233800dd7a4704cea8d887c7128)